src/IssueViz.Security/Hotspots/HotspotsList/ViewModels/HotspotsControlViewModel.cs

```diff
@@ -229,24 +238,33 @@ private void OnActiveDocumentChanged(object sender, ActiveDocumentChangedEventAr
             RefreshFiltering();
         }
 
-        private void RefreshFiltering() => RaisePropertyChanged(nameof(Hotspots));
+        private void RefreshFiltering()
+        {
+            UpdateFilteredHotspots();
+            RaisePropertyChanged(nameof(Hotspots));
+        }
 
-        private ObservableCollection<IHotspotViewModel> GetFilteredHotspots()
+        private void UpdateFilteredHotspots()
         {
             filteredHotspots.Clear();
-            var currentHotspots = GetHotspotsFilteredByLocationFilter();
-            currentHotspots.ToList().ForEach(filteredHotspots.Add);
-
-            return filteredHotspots;
+            var hotspotsToShow = GetHotspotsFilteredByLocationFilter(hotspots.ToList());
+            hotspotsToShow = GetHotspotsFilteredByPriorityFilter(hotspotsToShow.ToList());
```

Comment:
do we need to `.ToList()` twice here?

Reply:
They're both private methods that return IEnumerable, there's no need to be so defensive with the implementation. I think they could just accept IEnumerable themselves and only the third ToList is needed (line 252)